numpy/f2py/tests/test_quoted_character.py

```diff
@@ -28,3 +33,97 @@ class TestQuotedCharacter(util.F2PyTest):
                         reason='Fails with MinGW64 Gfortran (Issue #9673)')
     def test_quoted_character(self):
         assert_equal(self.module.foo(), (b"'", b'"', b';', b'!', b'(', b')'))
+
+@pytest.mark.parametrize("extra_args", [
+    # extra_args can be a list as of gh-11937
+    (['--noopt', '--debug']),
+    # test for string as well, using the same
+    # fcompiler options
+    ('--noopt --debug'),
+    ])
+def test_f2py_init_compile(extra_args):
+    # flush through the f2py __init__
+    # compile() function code path
+    # as a crude test for input handling
+    # following migration from exec_command()
+    # to subprocess.check_output() in gh-11937
+
+    fsource =  '''
+             integer function foo()
+             foo = 10 + 5
+             return
+             end
+             '''
```

Comment:
Is the column alignment of this fortran important? If so, would probably be good to add a comment to that effect.